src/main/java/com/mojang/brigadier/StringReader.java

```diff
@@ -76,6 +78,16 @@ public char peek(final int offset) {
         return string.charAt(cursor + offset);
     }
 
+    @Override
+    public boolean isNext(char c) {
+        return canRead() && peek() == c;
+    }
+
+    @Override
+    public boolean isNext(Predicate<Character> predicate) {
+        return canRead() && predicate.test(peek());
+    }
```

Comment:
Would it make sense to implement these as `default` methods of the interface instead?
(Though assuming no or only a few users implement their own `ImmutableStringReader` it likely makes no difference)